pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -284,7 +280,8 @@ func (pl *InterPodAffinity) AddPod(ctx context.Context, cycleState *framework.Cy
 	if err != nil {
 		return framework.NewStatus(framework.Error, err.Error())
 	}
-	state.updateWithPod(podToAdd, nodeInfo.Node(), 1)
+	updatedPodInfo := framework.NewPodInfo(podToAdd)
```

Comment:
This PR is just moving the TODO from `updateWithPod` to here, so not really solving anything.

Reply:
Yep. Maybe we can change the below interfaces from `*v1.Pod` to `*framework.PodInfo`, so we can construct PodInfo less in func like `selectVictimsOnNode`. But I not sure it is worth to break the consistency in `PluginsRunner` interface to achieve that. What are your original thoughts? @ahg-g  
 https://github.com/kubernetes/kubernetes/blob/68b908d78502e58333021736e5f617cf301fdc2d/pkg/scheduler/framework/interface.go#L584-L587

Reply:
we are looping the pods on the nodes here, pod info should be different with each other. Cannot see any improvement even the interface is changed to use `podInfo`,

Reply:
> we are looping the pods on the nodes here, pod info should be different with each other. Cannot see any improvement even the interface is changed to use `podInfo`,

`NewPodInfo` does some parsing. If we can use the one that is already stored, we save the time of parsing.

Reply:
We will need to do a couple of other changes as well to completely avoid creating podInfo unnecessarily: (1) `NodeInfo.Add/RemovePod` will need to be changed to accept PodInfo instead of Pod, and (2) [nominated pods](https://github.com/kubernetes/kubernetes/blob/989b2fd3715d01a7757e891de2a17de5a5c2cc91/pkg/scheduler/core/generic_scheduler.go#L383) will need to be tracked as PodInfos rather than pods.

I am ok with updating the interfaces for both RunPreFilterExtensionAdd/RemovePod and NodeInfo.Add/RemovePod. But lets do them in separate PRs.